pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -232,30 +254,53 @@ func (p *K8sPlugin) readManifestFiles() error {
 
 func (p *K8sPlugin) switchdevServiceStateUpdate() error {
 	// Check switchdev service
-	swdService, err := p.serviceManager.ReadService(p.switchdevService.Path)
+	swdService, err := p.serviceManager.ReadService(p.switchdevBeforeNMService.Path)
 	if err != nil {
 		if !os.IsNotExist(err) {
 			return err
 		}
 		// service not exists
-		p.updateTarget.switchdevService = true
+		p.updateTarget.switchdevBeforeNMService = true
 	} else {
-		needChange, err := service.CompareServices(swdService, p.switchdevService)
+		needChange, err := service.CompareServices(swdService, p.switchdevBeforeNMService)
 		if err != nil {
 			return err
 		}
-		p.updateTarget.switchdevService = needChange
+		p.updateTarget.switchdevBeforeNMService = needChange
+	}
+	swdService, err = p.serviceManager.ReadService(p.switchdevAfterNMService.Path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		// service not exists
+		p.updateTarget.switchdevAfterNMService = true
+	} else {
+		needChange, err := service.CompareServices(swdService, p.switchdevAfterNMService)
+		if err != nil {
+			return err
+		}
+		p.updateTarget.switchdevAfterNMService = needChange
 	}
 
 	// Check switchdev run script
-	data, err := ioutil.ReadFile(path.Join(chroot, p.switchdevRunScript.Path))
+	data, err := ioutil.ReadFile(path.Join(chroot, p.switchdevBeforeNMRunScript.Path))
```

Comment:
we could avoid some duplicate logic here if we defined a method: `bool ShouldUpdateServiceScript(Service)`

and call it for both service scripts

Reply:
Done, I used `isSwitchdevScriptNeedUpdate` as function name to be consistent with existing `isSystemServiceNeedUpdate`.